ls/ls.go

```diff
@@ -141,3 +152,100 @@ func (l *LS) showDirStructure() error {
 	fmt.Fprintln(l.StdOut, p)
 	return nil
 }
+
+func (l *LS) listFileIndex() error {
+	dirs, err := os.ReadDir(l.Dir)
+	if err != nil {
+		return err
+	}
+
+	for _, file := range dirs {
+		fileInfo, err := os.Stat(file.Name())
+		if err != nil {
+			return err
+		}
+
+		fmt.Fprintln(l.StdOut, fileInfo)
+	}
+
+	return nil
+}
+
+func (l *LS) sortFilesByTime() error {
+	var modTimes []string
+	var sortedFiles []string
+	dirs, err := os.ReadDir(l.Dir)
+	if err != nil {
+		return err
+	}
+
+	for _, file := range dirs {
+		fileStat, err := os.Stat(file.Name())
+		if err != nil {
+			return err
+		}
+		modTimes = append(modTimes, fileStat.ModTime().String())
+	}
+
+	sort.Sort(sort.Reverse(sort.StringSlice(modTimes)))
+
+	for _, v := range modTimes {
+		for _, file := range dirs {
+			fileStat, err := os.Stat(file.Name())
+			if err != nil {
+				return nil
+			}
+			if fileStat.ModTime().String() == v {
+				sortedFiles = append(sortedFiles, fileStat.Name())
+			}
+		}
+	}
+
+	for _, v := range sortedFiles {
+		fileStat, err := os.Stat(v)
+
+		if err != nil {
+			return err
+		}
+
+		fileName := fileStat.Name()
+		year := fileStat.ModTime().UTC().Year()
+		month := fileStat.ModTime().UTC().Month()
+		day := fileStat.ModTime().UTC().Day()
+		hour := fileStat.ModTime().UTC().Hour()
+		minute := fileStat.ModTime().UTC().Minute()
+		second := fileStat.ModTime().UTC().Second()
+		timeFormat := formatTime(hour, minute, second)
+		dateFormat := formatDate(year, int(month), day)
+
+		displayFormat := fmt.Sprintf("%s Date: %s Time: %s", fileName, dateFormat, timeFormat)
+
```

Comment:
should we remove the empty space, 
**_before_**
displayFormat := fmt.Sprintf("%s Date: %s Time: %s", fileName, dateFormat, timeFormat)

fmt.Fprintln(l.StdOut, displayFormat)

**_suggested_** 
displayFormat := fmt.Sprintf("%s Date: %s Time: %s", fileName, dateFormat, timeFormat)
fmt.Fprintln(l.StdOut, displayFormat)

Reply:
Sure why not. Thanks